sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2169,12 +2169,29 @@ class Analyzer(override val catalogManager: CatalogManager)
                         unbound, arguments, unsupported)
                   }
 
+                  if (bound.inputTypes().length != arguments.length) {
+                    throw QueryCompilationErrors.v2FunctionInvalidInputTypeLengthError(
+                      bound, arguments)
+                  }
+
+                  val castedArguments = arguments.zip(bound.inputTypes()).map { case (arg, ty) =>
+                    if (arg.dataType != ty) {
+                      if (Cast.canCast(arg.dataType, ty)) {
```

Comment:
This is a bit hacky to do type coercion manually, instead of using the existing type coercion framework. e.g. how do you support ANSI type coercion here? See `AnsiTypeCoercion` for more details.

My proposal is to let `Invoke`/`StaticInvoke` extend `ImplicitCastInputTypes`, and leverage the existing type coercion framework.

Reply:
Good point! I totally ignored `AnsiCast`. Let me try out your suggestion. Thanks.